src/renderer/components/database-diagram.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component, PropTypes } from 'react';
+import joint from 'jointjs';
+
+require('./jointjs-diagram-table.js');
+require('./jointjs-diagram-table-cell.js');
+require('jointjs/dist/joint.min.css');
+
+
+export default class DatabaseDiagram extends Component {
+  static propTypes = {
+    tables: PropTypes.array,
+    columnsByTable: PropTypes.object,
+    links: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.graph = new joint.dia.Graph();
+  }
+
+  componentDidMount() {
```

Comment:
I guess there is too much stuff going on in the componentDidMount. If you could break down in smaller functions would be really great. Maybe everything that you have included a comment can become a function.